pkg/render/render_test.go

```diff
@@ -0,0 +1,114 @@
+package render
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("RenderTemplate", func() {
+	Context("when a valid YAML manifest is given as a parameter", func() {
+		It("should successfully render", func() {
+			renderData := MakeRenderData()
+
+			renderedFromYAML, err := RenderTemplate("testdata/simple.yaml", &renderData)
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(renderedFromYAML).To(HaveLen(1))
+			expected := `
+{
+	"apiVersion": "v1",
+	"kind": "Pod",
+	"metadata": {
+		"name": "busybox1",
+		"namespace": "ns"
+	},
+	"spec": {
+		"containers": [
+			{
+  				"image": "busybox"
+			}
+		]
+	}
+}
+`
+			Expect(renderedFromYAML[0].MarshalJSON()).To(MatchJSON(expected))
+		})
+	})
+
+	Context("when a identical YAML and JSONs manifest are given as a parameters", func() {
+		It("should return identical objects", func() {
```

Comment:
Isn't this identical to previous test  more or less ?

Reply:
This one is making sure JSON works fine. Would you write it differently?

Reply:
Actually, I should just read JSON and compare it to expected struct

Reply:
Done